tests/main/umask/task.yaml

```diff
@@ -0,0 +1,15 @@
+summary: directories created by snap-confine are immune from umask
+details: |
+    The setting of umask is inherited across the snap-{run,confine,exec} chain
+    but does not hinder execution of snap-confine itself.
+prepare: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-sh
+restore: |
+    snap remove test-snapd-sh
+    rm -rf ~/snap
+execute: |
+    ( umask 777 && test-snapd-sh -c /bin/true )
+    ( umask 777 && test-snapd-sh -c 'touch "$SNAP_USER_COMMON"/canary' )
+    test "$(stat -c %a "$HOME"/snap/test-snapd-sh/common/canary)" = 0
```

Comment:
I'd probably try with some ridiculous umask like 000 too, just to make sure that we're not closing it down unnecessarily.

Reply:
Done